neoutils/connection.go

```diff
@@ -37,6 +41,18 @@ func Connect(neoURL string, conf *ConnectionConfig) (NeoConnection, error) {
 		conf = DefaultConnectionConfig()
 	}
 
+	if !conf.BackgroundConnect {
+		return connectDefault(neoURL, conf)
+	} else {
+		f := func() (NeoConnection, error) {
+			return connectDefault(neoURL, conf)
+		}
+		return connectAuto(neoURL, f, 30*time.Second)
```

Comment:
It was pretty randomly chosen. Too short means causing noise and load.  Too long means more downtime than needed.

We can easily tweak in future, adding a config option if that makes sense.